src/com/facebook/buck/maven/Resolver.java

```diff
@@ -439,25 +584,38 @@ private String buildKey(Artifact artifact) {
         ':' + artifact.getClassifier();
   }
 
-  public static void main(String[] args) throws RepositoryException, IOException {
-    if (args.length < 5) {
+  public static void main(String[] args) throws Throwable {
+    if (args.length < 4) {
       System.err.println("Usage: java -jar resolver.jar buck-repo third-party " +
-              "maven-local-repo maven-url junit:junit:jar:4.12...");
+          "maven-local-repo maven-url... -- junit:junit:jar:4.12...");
+      System.err.println("   or: java -jar resolver.jar buck-repo third-party " +
+          "maven-local-repo path-to-pom.xml");
       System.exit(1);
     }
 
     Path buckRepoRoot = Paths.get(args[0]);
     Path thirdParty = Paths.get(args[1]);
     Path m2 = Paths.get(args[2]);
-    String mavenCentral = args[3];
-    String[] coords = Arrays.copyOfRange(args, 4, args.length);
-
-    new Resolver(
+    Resolver resolver = new Resolver(
         buckRepoRoot,
         thirdParty,
-        m2,
-        mavenCentral)
-        .resolve(coords);
+        m2);
+
+    if (args.length == 4) {
+      resolver.loadPomFile(Paths.get(args[3]));
+    } else {
+      List<String> repos = new ArrayList<>();
+      int i;
```

Comment:
`i` is used outside `for` loop, so it's better to name it accordingly. Like `indexOfShellPassThroughArgListSeparator` or how ever you'd call this special argument.